src/clients/kibisis/client.ts

```diff
@@ -417,69 +473,45 @@ class KibisisClient extends BaseClient {
 
   async signTransactions(
     connectedAccounts: string[],
-    transactions: Uint8Array[],
+    transactionsOrTransactionGroups: Uint8Array[] | Uint8Array[][],
     indexesToSign?: number[],
     returnGroup = true
   ): Promise<Uint8Array[]> {
     await this.refreshSupportedMethods() // refresh the supported methods
 
-    const txns: Arc0001SignTxns[] = await Promise.all(
-      transactions.map<Promise<Arc0001SignTxns>>(async (value, index) => {
-        const decodedTxn = this.algosdk.decodeObj(value) as
-          | DecodedTransaction
-          | DecodedSignedTransaction
-        const isSigned = !!(decodedTxn as DecodedSignedTransaction).txn
-        const sender = this.algosdk.encodeAddress(
-          isSigned
-            ? (decodedTxn as DecodedSignedTransaction).txn.snd
-            : (decodedTxn as DecodedTransaction).snd
-        )
-        const accountInfo = await this.getAccountInfo(sender)
-        const authAddr = accountInfo['auth-addr']
-        const txn = this.convertBytesToBase64(
-          this.algosdk.decodeUnsignedTransaction(value).toByte()
-        )
-
-        // if the transaction is signed, instruct the provider not to sign by providing an empty signers array
-        if (isSigned) {
-          return {
-            txn: this.convertBytesToBase64(
-              this.algosdk.decodeSignedTransaction(value).txn.toByte()
-            ),
-            signers: [],
-            ...(authAddr && { authAddr })
-          }
-        }
-
-        // if the sender is not authorized or the index has not been included in the to be signed indexes, instruct the provider not to sign by providing an empty signers array
-        if (
-          !connectedAccounts.includes(sender) ||
-          (indexesToSign && !indexesToSign.includes(index))
-        ) {
-          return {
-            txn,
-            signers: [],
-            ...(authAddr && { authAddr })
-          }
-        }
+    // TODO: the below disable/ignore is necessary as the reduce function throws a TS error for union types (https://github.com/microsoft/TypeScript/issues/36390), however, these can be removed when typescript is updated to 5.2.2+, as the issue has been fixed
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const transactions: ARC0001SignTxns[] = transactionsOrTransactionGroups.reduce<
+      ARC0001SignTxns[]
+    >((acc: ARC0001SignTxns[], currentValue: Uint8Array | Uint8Array[], index: number) => {
+      const toSign = indexesToSign && indexesToSign.includes(index)
+
+      // if an element is an array, concatenate each mapped element as we want a flat (one-dimensional) array to send to the wallet
+      if (Array.isArray(currentValue)) {
+        return [
+          ...acc,
+          ...currentValue.map((value) =>
+            this.mapRawTransactionToARC0001Transaction(value, connectedAccounts, toSign)
```

Comment:
Your `mapRawTransactionToARC0001Transaction` method is async and returns a promise. Since this reduce function is not async and you're not using await to call your method, it's returning an unresolved `Promise<ARC0001SignTxns>`, not the `ARC0001SignTxns` you want.

The result is the following error:

```
Uncaught DOMException: Promise object could not be cloned.
```

Your previous solution used `await Promise.all` to map over the transactions. You'll probably need to do something similar here with the reduce function.

Reply:
A simpler solution might be just to `flatMap` the nested array then process it the same as before. Here is the utility function I use in v3 (which also accepts `algosdk.Transaction` objects) 

https://github.com/TxnLab/use-wallet/blob/v3/packages/use-wallet/src/utils.ts#L94